File: app/src/main/java/kagura/project/com/a8/LoadJson.java
package kagura.project.com.a8;

import android.content.Context;

import java.io.IOException;
import java.io.InputStream;

public class LoadJson {

    public String loadJSONFromAsset(Context c, String file) {
        String json;
        try {
            InputStream is = c.getAssets().open(file + ".json");
            int size = is.available();
            byte[] buffer = new byte[size];

            //noinspection ResultOfMethodCallIgnored
            is.read(buffer);
            is.close();

            json = new String(buffer, "UTF-8");
        } catch (IOException ex) {
            ex.printStackTrace();
            return null;
        }
        return json;
    }
}

File: app/src/main/java/kagura/project/com/fruitlegumes/LoadJson.kt
package kagura.project.com.fruitlegumes

import android.content.Context
import android.util.Log
import java.io.IOException
import java.nio.charset.Charset

object LoadJson {
    @JvmStatic
    fun loadJSONFromAsset(c: Context, file: String): String? {
        val json: String
        try {
            val `is` = c.assets.open("$file.json")
            val size = `is`.available()
            val buffer = ByteArray(size)
            `is`.read(buffer)
            `is`.close()
            json = String(buffer, Charset.forName("UTF-8"))
            Log.i("json", json)
        } catch (ex: IOException) {
            ex.printStackTrace()
            return null
        }
        return json
    }
}